lib/SimpleSAML/Kernel.php

```diff
@@ -1,170 +1,69 @@
-        $configuration = Configuration::getInstance();
-        $cachePath = $configuration->getString('tempdir') . '/cache/' . $this->module;
-
-        if (0 === strpos($cachePath, '/')) {
-            return $cachePath;
-        }
-
-        return $configuration->getBaseDir() . '/' . $cachePath;
-    }
-
-
-    /**
-     * @return string
-     */
-    public function getLogDir(): string
+    public function registerBundles()
     {
-        $configuration = Configuration::getInstance();
-        $loggingPath = $configuration->getString('loggingdir');
-
-        if (0 === strpos($loggingPath, '/')) {
-            return $loggingPath;
+        $contents = require $this->getProjectDir().'/config/modules.php';
+        foreach ($contents as $class => $envs) {
+            if ($envs[$this->environment] ?? $envs['all'] ?? false) {
+                yield new $class();
+            }
         }
-
-        return $configuration->getBaseDir() . '/' . $loggingPath;
     }
 
-
-    /**
-     * @return array
-     */
-    public function registerBundles(): array
-    {
-        return [
-            new FrameworkBundle(),
-        ];
-    }
-
-
-    /**
-     * Get the module loaded in this kernel.
-     *
-     * @return string
-     */
-    public function getModule(): string
-    {
-        return $this->module;
-    }
-
-
-    /**
-     * Configures the container.
-     *
-     * @param ContainerBuilder $container
-     * @param LoaderInterface $loader
-     * @return void
-     */
-    protected function configureContainer(ContainerBuilder $container, LoaderInterface $loader): void
+    protected function configureRoutes(RouteCollectionBuilder $routes)
     {
-        $configuration = Configuration::getInstance();
-        $baseDir = $configuration->getBaseDir();
-        $loader->load($baseDir . '/routing/services/*' . self::CONFIG_EXTS, 'glob');
-        $confDir = Module::getModuleDir($this->module) . '/routing/services';
-        if (is_dir($confDir)) {
-            $loader->load($confDir . '/**/*' . self::CONFIG_EXTS, 'glob');
+        foreach ($this->bundles as $bundle) {
+            if ($bundle instanceof \SimpleSAML\Modules\Module) {
+                $baseDir = $bundle->getPath() . '/Resources/config';
+                $name = strtolower($bundle->getName());
+                $prefix = substr(
+                    $name,
+                    0,
+                    strpos($name, 'module'));
+
+                if (is_dir($baseDir)) {
+                    $routes->import($baseDir . '/routes' . self::CONFIG_EXTS, $prefix, 'glob');
+                }
+            }
         }
 
-        $container->loadFromExtension('framework', [
-            'secret' => Configuration::getInstance()->getString('secretsalt'),
-        ]);
-
-        $this->registerModuleControllers($container);
-    }
+        $confDir = __DIR__ . '/Resources/config';
 
+        $routes->import($confDir.'/{routes}/'.$this->environment.'/**/*'.self::CONFIG_EXTS, '/', 'glob');
```

Comment:
Load routes required by simplesaml to work. For example the route required to have the debug toolbar.